integration-tests/docker/test_env/test_env_builder.go

```diff
@@ -89,41 +98,64 @@ func (b *CLTestEnvBuilder) buildNewEnv(cfg *TestEnvConfig) (*CLClusterTestEnv, e
 	if cfg != nil {
 		te, err = NewTestEnvFromCfg(cfg)
 		if err != nil {
-			return te, err
+			return nil, err
 		}
 	} else {
 		te, err = NewTestEnv()
 		if err != nil {
-			return te, err
+			return nil, err
 		}
 	}
 
 	if b.hasLogWatch {
 		lw, err := logwatch.NewLogWatch(nil, nil)
 		if err != nil {
-			return te, err
+			return nil, err
 		}
 		te.LogWatch = lw
 	}
 
 	if b.hasMockServer {
 		err := te.StartMockServer()
 		if err != nil {
-			return te, err
+			return nil, err
 		}
 		err = te.MockServer.SetExternalAdapterMocks(b.externalAdapterCount)
 		if err != nil {
-			return te, err
+			return nil, err
 		}
 	}
-
-	if b.hasGeth {
-		err := te.StartGeth()
+	networkConfig := networks.SelectedNetwork
+	var internalDockerUrls InternalDockerUrls
+	if b.hasGeth && networkConfig.Simulated {
+		networkConfig, internalDockerUrls, err = te.StartGeth()
 		if err != nil {
-			return te, err
+			return nil, err
 		}
 	}
 
+	bc, err := blockchain.NewEVMClientFromNetwork(networkConfig)
+	if err != nil {
+		return nil, err
+	}
+	// Get blockchain.EthereumClient as this is the only possible client for Geth
```

Comment:
@iljapavlovs This comment is no longer true, we should remove this code and use `blockchain.EVMClient`.
Currently, in both testnet and Geth cases we have `te.EthClient` as `*blockchain.EthereumClient` and it won't work for different networks.
 Please use `blockchain.EVMClient` in all the cases if possible, if using a concrete type is easier it should be used only when `hasGeth` flag is set.

Reply:
fixed